rpcs3/util/atomic.cpp

```diff
@@ -1,15 +1,19 @@
 ﻿#include "atomic.hpp"
 
-// TODO: something for other platforms
-#if defined(__linux__) || !defined(_WIN32)
+#if defined(__linux__)
 #define USE_FUTEX
 #endif
 
+#if !defined(USE_FUTEX) && !defined(_WIN32)
+#define USE_STD
+#endif
+
```

Comment:
With these changes, this whole section is equivalent to
```c++
#if defined(__linux__)
#define USE_FUTEX
#elif !defined(_WIN32)
#define USE_STD
#endif
```

I recommend writing this instead, as it is very confusing as written now, and could lead to bugs like the below.

Reply:
You can use USE_FUTEX and USE_STD on any platform (there is reference futex implementation code).

Reply:
I think it makes more sense to let anyone who wants to add it later make the necessary changes, rather than have stuff the way it is now. If we start guarding against hypothetical code, that's quite a rabbit hole...

Reply:
I can change it a little, it's not a big deal.